File: admins/src/main/java/com/github/admins/controllers/impl/CommentController.java
package com.github.admins.controllers.impl;

import com.github.admins.controllers.ICommentController;
import com.github.admins.dto.CommentDto;
import com.github.admins.exceptions.NotFound;
import com.github.admins.services.ICommentService;
import com.github.admins.utils.Logging;
import com.netflix.hystrix.contrib.javanica.annotation.HystrixCommand;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import static com.github.admins.utils.TransferObj.fromComment;

@RestController
@RequestMapping(path = "/v1/comment")
@RequiredArgsConstructor
public class CommentController implements ICommentController {

    private final ICommentService commentService;

    @Override
    @HystrixCommand
    @Logging(isTime = true, isReturn = false)
    public CommentDto findById(Long id) {
        return fromComment(this.commentService.readById(id)
                .orElseThrow(NotFound::new));
    }

    @Override
    @HystrixCommand
    @Logging(isTime = true, isReturn = false)
    public void remove(Long id) {
        this.commentService.delete(id);
    }
}

File: server-mocks/src/main/kotlin/com/github/server/mocks/admins/controllers/CommentController.kt
package com.github.server.mocks.admins.controllers

import com.github.server.mocks.admins.payload.Comment
import com.github.server.mocks.admins.services.impl.CommentService
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.*

@RestController
class CommentController(private val commentService: CommentService) {

    var log: Logger = LoggerFactory.getLogger(this.javaClass)

    @PostMapping(
            "/v1/comments/"
    )
    fun create(@RequestBody comment: Comment): Comment {
        log.info("Enter: {}", comment)
        return this.commentService.create(comment)
    }

    @GetMapping(
            "/v1/comments/info/{id}"
    )
    fun readById(@PathVariable id: Long): Comment {
        log.info("Enter: read by id {}", id)
        return this.commentService.readById(id)
    }

    @DeleteMapping(
            "/v1/comments/edit/{id}"
    )
    fun remove(@PathVariable id: Long) {
        log.info("Enter: remove product with id {}", id)
    }

}